packages/playwright-test/src/matchers/toMatchSnapshot.ts

```diff
@@ -289,6 +290,10 @@ export async function toHaveScreenshot(
       testInfo, testInfo._screenshotPath.bind(testInfo), 'png',
       testInfo.project.expect?.toHaveScreenshot || {},
       nameOrOptions, optOptions);
+  if (!helper.snapshotPath.toLowerCase().endsWith('.png'))
+    throw new Error(`Screenshot name "${path.basename(helper.snapshotPath)}" must have '.png' extension`);
+  if (Buffer.isBuffer(pageOrLocator) || (typeof pageOrLocator === 'string'))
```

Comment:
There is an `expectType` helper in `playwright-test/src/util.ts`.

Reply:
check for page or locator explicitly